integration/helper_test.go

```diff
@@ -391,7 +393,249 @@ func createTestOASApi(ctx context.Context, ns string, c *envconf.Config, tykOASD
+	}
+
+	if oasApi.Status.LatestTransaction.Status != expectations.status {
+		return fmt.Errorf(
+			"invalid LatestTransaction.Status in TykOasApiDefinition status, TykOasApiDefinition Status: %+v",
+			oasApi.Status,
+		)
+	}
+
+	if oasApi.Status.ListenPath != expectations.listenPath {
+		return fmt.Errorf(
+			"invalid ListenPath in TykOasApiDefinition status, TykOasApiDefinition Status: %+v", oasApi.Status,
+		)
+	}
+
+	return nil
+}
+
+// checkOasApisReferExistingConfigMaps checks if each member of oasApis refers to the existing configMaps.
+func checkOasApisReferExistingConfigMaps(oasApis []v1alpha1.TykOasApiDefinition, configMaps []v1.ConfigMap) bool {
+	type mapKey string
+	key := func(ns, name string) mapKey {
+		return mapKey(ns + "/" + name)
+	}
+
+	configMapSet := make(map[mapKey]bool)
+	for i := range configMaps {
+		configMapSet[key(configMaps[i].Namespace, configMaps[i].Name)] = true
+	}
+
+	for i := range oasApis {
+		k := key(oasApis[i].Spec.TykOAS.ConfigmapRef.Namespace, oasApis[i].Spec.TykOAS.ConfigmapRef.Name)
+		if _, exists := configMapSet[k]; !exists {
+			return false
+		}
+	}
+
+	return true
+}
+
+// validateOasAuthentication checks for if authentication related configurations are inherited from templates.
+// It takes two byte arrays; the first represents OAS API returned from Tyk Client and the second represents
+// template ConfigMap's data.
+func validateOasAuthentication(tykOasApi, tplCmData []byte) error {
+	tykAuthType, err := controllers.GetString(tykOasApi, controllers.ServerAuthenticationBaseIdentityProviderKeys...)
+	if err != nil {
+		return fmt.Errorf("failed to parse 'server.authentication' from OasApiDefinition fetched from Tyk, err: %v", err)
+	}
+
+	expectedAuthType, err := controllers.GetString(tplCmData, controllers.ServerAuthenticationBaseIdentityProviderKeys...)
+	if err != nil {
+		return fmt.Errorf("failed to parse 'info.name' from  fetched from Tyk, err: %v", err)
+	}
+
+	if tykAuthType != expectedAuthType {
+		return fmt.Errorf("unexpected OAS Auth Type, expected %v got %v", expectedAuthType, tykAuthType)
+	}
+
+	return nil
+}
+
+// validateInfoName validates if the given byte array, which represents OAS API from Tyk in JSON format, has
+// expected info name given as argument to the function.
+func validateInfoName(tykOasApi []byte, expectedInfoName string) error {
+	tykOasApiName, err := controllers.GetString(tykOasApi, controllers.InfoNameKeys...)
+	if err != nil {
+		return fmt.Errorf("failed to parse 'info.name' from OasApiDefinition fetched from Tyk, err: %v", err)
+	}
+
+	if tykOasApiName != expectedInfoName {
+		return fmt.Errorf("unexpected OAS API Name, expected %v got %v", expectedInfoName, tykOasApiName)
+	}
+
+	return nil
+}
+
+// validateListenPath validates if the given byte array, which represents OAS API from Tyk in JSON format, has
+// expected listen path given as argument to the function.
+func validateListenPath(tykOasApi []byte, expectedListenPath string) error {
```

Comment:
`validateInfoName`, `validateListenPath`, `validateUpstreamURL` are all validating a string.
Maybe you could create a generic function that accepts the key instead of duplicating logic.

Reply:
I thought that as these functions are used in tests, it might be beneficial to make them more verbose, including error messages and function names - that's why I created separate functions. However, if you believe that a generic function to handle validation logic would be better, I am happy to refactor accordingly.

Reply:
I would like to avoid code duplication as much as possible.
If you could make it generic it's great.